### PR TITLE
Closes #70: Pig example (id extractor) on spark

### LINKED ISSUE #70: Write a spark equivalent of core/examples/pig/basic/workflow.xml
With the pig node changed to spark.

### PR DESCRIPTION
**Pull request by madryk**

### CODE REVIEW

File: iis-workflows/iis-workflows-core-examples/src/test/java/eu/dnetlib/iis/core/examples/spark/PersonIdExtractorWorkflowTest.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import eu.dnetlib.iis.core.OozieWorkflowTestConfiguration;
+import eu.dnetlib.iis.core.TestsIOUtils;
+import eu.dnetlib.iis.core.WorkflowTestResult;
+
+/**
+ * 
+ * @author madryk
+ *
+ */
+public class PersonIdExtractorWorkflowTest extends AbstractOozieWorkflowTestCase {
```
Comment (lukdumi) [inline iis-workflows/iis-workflows-core-examples/src/test/java/eu/dnetlib/iis/core/examples/spark/PersonIdExtractorWorkflowTest.java]: I am not sure if surefire sees the @Category defined in super class.

File: iis-workflows/iis-workflows-core-examples/src/main/java/eu/dnetlib/iis/core/examples/spark/PersonIdExtractor.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import org.apache.commons.lang.StringUtils;
+import org.apache.spark.SparkConf;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+
+import com.clearspring.analytics.util.Preconditions;
+
+/**
+ * Service for extracting ids from person saved in comma separated file and
+ * storing results to the given directory
```
Comment (lukdumi) [inline iis-workflows/iis-workflows-core-examples/src/main/java/eu/dnetlib/iis/core/examples/spark/PersonIdExtractor.java]: maybe: ... in the given...

File: iis-workflows/iis-workflows-core-examples/src/test/resources/eu/dnetlib/iis/core/examples/spark/person_id_extractor/oozie_app/workflow.xml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        <error to="fail"/>
+    </action>
+    <action name="spark_node">
+        <spark xmlns="uri:oozie:spark-action:0.1">
+            <job-tracker>${jobTracker}</job-tracker>
+            <name-node>${nameNode}</name-node>
+            <prepare>
+				<delete path="${nameNode}${workingDir}/output" />
+				<mkdir path="${nameNode}${workingDir}/output" />
+			</prepare>
```
Comment (lukdumi) [inline iis-workflows/iis-workflows-core-examples/src/test/resources/eu/dnetlib/iis/core/examples/spark/person_id_extractor/oozie_app/workflow.xml]: something with formatting

File: iis-workflows/iis-workflows-core-examples/src/test/java/eu/dnetlib/iis/core/examples/spark/PersonIdExtractorWorkflowTest.java (hunk 1 above)
Comment (madryk) [inline iis-workflows/iis-workflows-core-examples/src/test/java/eu/dnetlib/iis/core/examples/spark/PersonIdExtractorWorkflowTest.java]: Surefire in version that we are using will inherit category from base class. I checked it.
http://stackoverflow.com/questions/27248795/why-does-junit-ignore-category-annotations-on-base-classes

Comment (madryk) [inline iis-workflows/iis-workflows-core-examples/src/test/java/eu/dnetlib/iis/core/examples/spark/PersonIdExtractorWorkflowTest.java]: I've added `@Category` for `AbstractOozieWorkflowTestCase`. Earlier it was not annotated at all.

File: iis-workflows/iis-workflows-core-examples/src/main/java/eu/dnetlib/iis/core/examples/spark/PersonIdExtractor.java (hunk 2 above)
Comment (madryk) [inline iis-workflows/iis-workflows-core-examples/src/main/java/eu/dnetlib/iis/core/examples/spark/PersonIdExtractor.java]: ok

File: iis-workflows/iis-workflows-core-examples/src/test/resources/eu/dnetlib/iis/core/examples/spark/person_id_extractor/oozie_app/workflow.xml (hunk 3 above)
Comment (madryk) [inline iis-workflows/iis-workflows-core-examples/src/test/resources/eu/dnetlib/iis/core/examples/spark/person_id_extractor/oozie_app/workflow.xml]: ok. I will correct it.

**Pull request closed without merging**